variables.tf

```diff
@@ -0,0 +1,364 @@
+  type        = string
+  default     = ""
+}
+
+variable "user_status" {
+  description = "The status of the resource. Valid values: Unlocked, Locked."
+  type        = string
+  default     = "Unlocked"
+}
+
+#alicloud_ecd_policy_group
+variable "create_policy_group" {
+  description = "Controls if ecd policy group should be created"
+  type        = bool
+  default     = false
+}
+
+variable "policy_group_name" {
+  description = "The name of policy group."
+  type        = string
+  default     = ""
+}
+
+variable "policy_clipboard" {
+  description = "The clipboard policy. Valid values: off, read, readwrite."
+  type        = string
+  default     = "off"
+}
+
+variable "policy_local_drive" {
+  description = " Local drive redirect policy. Valid values: readwrite, off, read."
+  type        = string
+  default     = "off"
+}
+
+variable "policy_usb_redirect" {
+  description = "The usb redirect policy. Valid values: off, on."
+  type        = string
+  default     = "off"
+}
+
+variable "policy_watermark" {
+  description = "The watermark policy. Valid values: off, on."
+  type        = string
+  default     = "off"
+}
+
+variable "authorize_access_policy_rules" {
+  description = "The rule of authorize access rule."
+  type        = map(string)
+  default     = {}
+}
+
+variable "authorize_security_policy_rules" {
+  description = "The policy rule."
+  type        = map(string)
+  default     = {}
+}
+
+variable "domain_list" {
+  description = "The list of domain."
+  type        = string
+  default     = ""
+}
+
+variable "html_access" {
+  description = "The access of html5. Valid values: off, on."
+  type        = string
+  default     = "off"
+}
+
+variable "html_file_transfer" {
+  description = "The html5 file transfer. Valid values: all, download, off, upload."
+  type        = string
+  default     = "off"
+}
+
+variable "visual_quality" {
+  description = "The quality of visual. Valid values: high, lossless, low, medium."
+  type        = string
+  default     = "low"
+}
+
+variable "watermark_transparency" {
+  description = "The watermark transparency. Valid values: DARK, LIGHT, MIDDLE."
+  type        = string
+  default     = "DARK"
+}
+
+variable "watermark_type" {
+  description = "The type of watemark. Valid values: EndUserId, HostName."
+  type        = string
+  default     = "HostName"
+}
+
+#alicloud_ecd_desktop
+variable "create_desktop" {
+  description = "Controls if ecd desktop should be created"
+  type        = bool
+  default     = false
+}
+
+variable "desktop_name" {
+  description = " The desktop name of the Desktop."
+  type        = string
+  default     = ""
+}
+
+variable "bundle_id" {
+  description = "The bundle id of the Desktop."
+  type        = string
+  default     = ""
+}
+
+variable "office_site_id" {
+  description = "The ID of the Simple Office Site."
+  type        = string
+  default     = ""
+}
+
+variable "policy_group_id" {
+  description = "The policy group id of the Desktop."
```

Comment:
这种需要传入 ID的情况，描述要写清楚，当 create xx 为true的时候，这个参数将被忽略，比如：
description = "The policy group id of the Desktop. It will be ignored when create_policy_group = true"

Reply:
ok